File: Binary Tree Postorder Traversal.cpp
#include<iostream>
#include<vector>
#include<stack>

using namespace std;

struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

void dfs(vector<int> &vec,TreeNode *root)
    {
        if(root->left!=NULL) dfs(vec,root->left);
        vec.push_back(root->val);
        if(root->right!=NULL) dfs(vec,root->right);
    }

vector<int> postorderTraversal(TreeNode *root)
    {
        vector<int> vec;
        stack<int> num;
        stack<TreeNode*> st;
        if(root!=NULL) st.push(root);
        TreeNode *node;
        while(!st.empty())
            {
                node=st.top();
                st.pop();
                if(node->left!=NULL) st.push(node->left);
                if(node->right!=NULL) st.push(node->right);
                num.push(node->val);
            }
        while(!num.empty())
            {
                vec.push_back(num.top());
                num.pop();
            }
        return vec;
    }

int main()
    {
        return 0;
    }

File: Python/Binary Tree Postorder Traversal.py
#!/usr/bin/env python

# Definition for a  binary tree node
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of integers
    def Traversal(self,root,res):
        if root != None:
            self.Traversal(root.left,res);
            self.Traversal(root.right,res);
            res.append(root.val)
        return
    def postorderTraversal(self, root):
        res=[]
        self.Traversal(root,res)
        return res